### PR TITLE
Don't log analytics in UI tests

### PR DESCRIPTION
**Pull request by vardges-stripe**
## Summary

Avoid logging exposures in unit/ui tests.
## Motivation

No need to mess up data with continues tests.
## Testing

See before my change I hit exposure logging and after I don't.
https://user-images.githubusercontent.com/92807969/228948350-f4111526-482e-4e5d-99c1-6a582f52054a.mov

### CODE REVIEW

File: StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift [hunk 1]
```diff
@@ -80,6 +83,19 @@ class FlowRouter {
...
+private extension FlowRouter {
+    static let isUnitOrUITest: Bool = {
+        #if targetEnvironment(simulator)
+            return NSClassFromString("XCTest") != nil
+                || ProcessInfo.processInfo.environment["UITesting"] != nil
+        #else
+            return false
+        #endif
+    }()
+}
```
[the remark this answers is not in the document]
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift:97]: I guess some folks are saying `NSClassFromString` does not work for UI tests

[the remark this answers is not in the document]
Comment (kgaidis-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift:97]: There is this too but debatable: https://stackoverflow.com/a/63447524/826435

**Review by kgaidis-stripe — DISMISSED** [review]
Solves the current problem

File: StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (vardges-stripe) [reply StripeFinancialConnections/StripeFinancialConnections/Source/Common/FlowRouter.swift:97]: Ahh good find, didn't realize it was there in analytics. yeah I agree it's stupid for us to keep this, we should just fix it in analytics for everyone.

**Comment by vardges-stripe** [comment]
@kgaidis-stripe I'll just fix analytics, it doesn't make sense to have this logic in many places.

**Pull request closed without merging**